std/internal/math/gammafunction.d

```diff
@@ -111,7 +111,14 @@ real gammaStirling(real x)
     }
     else {
         w = 1.0L + w * poly( w, SmallStirlingCoeffs);
-        y = pow( x, x - 0.5L ) / y;
+        if ( (floatTraits!(real).realFormat == RealFormat.ieeeDouble) && (x > 143.0L) ) {
```

Comment:
Yes. This code looks better for math. Furthermore we would have this functions as template in the future, so please replace `0.5L` for 64bit float with 0.5.